IO/ConsoleUtility.cs

```diff
@@ -0,0 +1,51 @@
+using System.Runtime.InteropServices;
+#if NET7_0_OR_GREATER
+using System.Runtime.Versioning;
+#endif
+
+namespace Messerli.IO;
+
+public static class ConsoleUtility
+{
+    /// <summary><para>Determines whether the current process is the last one attached to the console.
+    /// If this process is the last one, the console will be destroyed when the process exists.</para>
+    /// <para>This function is useful for determining if the console should
+    /// be kept open (e.g. by using <see cref="Console.ReadLine"/>).</para></summary>
+    /// <example><code>
+    /// using static Messerli.IO.ConsoleUtility;
+    ///
+    /// if (IsLastProcessAttachedToConsole())
+    /// {
+    ///     Console.WriteLine("Press Enter to continue...");
+    ///     Console.ReadLine();
+    /// }
+    /// </code></example>
+    /// <remarks>This API is only supported on windows only.</remarks>
+#if NET7_0_OR_GREATER
+    [SupportedOSPlatform("windows")]
+#endif
+    public static bool IsLastProcessAttachedToConsole()
+        => Windows.IsLastProcessAttachedToConsole();
```

Comment:
Hint to reviewer: I decided to deviate from our internal implementation here by requiring windows. This forces consumers to think about how they want to use the API instead of silently falling back to `false` on non-Windows platforms.